engines/python/setup/djl_python/rolling_batch/rolling_batch.py

```diff
@@ -168,34 +171,36 @@ def inference(self, input_data, parameters):
         """
         pass
 
-    def get_new_requests(self, input_data, parameters, batch_size):
-        new_requests = []
-        pending_req_len = len(self.pending_requests)
-        if batch_size > pending_req_len:
-            for i in range(pending_req_len, batch_size):
-                data = input_data[i]
-                params = parameters[i] if i < len(parameters) else {}
-                request = Request(self.req_id_counter, data, params)
-                self.pending_requests.append(request)
-                new_requests.append(request)
-                self.req_id_counter += 1
+    def get_new_requests(self, input_data, parameters):
+        init_batch_size = len(self.pending_requests)
+        for data, params in zip(input_data, parameters):
+            request = Request(self.req_id_counter, data,
+                              params if params else {})
+            self.pending_requests.append(request)
+            self.req_id_counter += 1
+
+            if self.max_batch_size is not None and len(
+                    self.pending_requests) == self.max_batch_size:
+                warnings.warn(
```

Comment:
Avoid using `warnings`, user can change logging configuration.
do you really want to just logging or you want to throw an exception?

Reply:
frontend should ensure the batch size is never exceed that max, did you ever seen this happen?